src/apps/testapps/testPolygonToCellsExperimental.c

```diff
@@ -598,6 +621,282 @@ SUITE(polygonToCells) {
         free(hexagons);
     }
 
+    TEST(polygonToCellsNullPolygon) {
+        int64_t numHexagons;
+        t_assertSuccess(H3_EXPORT(maxPolygonToCellsSizeExperimental)(
+            &nullGeoPolygon, 9, CONTAINMENT_CENTER, &numHexagons));
+        t_assert(numHexagons == 0, "got expected estimated size");
+        H3Index *hexagons = calloc(numHexagons, sizeof(H3Index));
+
+        t_assertSuccess(H3_EXPORT(polygonToCellsExperimental)(
+            &nullGeoPolygon, 9, CONTAINMENT_CENTER, hexagons));
+        int64_t actualNumIndexes = countNonNullIndexes(hexagons, numHexagons);
+
+        t_assert(actualNumIndexes == 0, "got expected polygonToCells size");
+        free(hexagons);
+    }
+
+    TEST(polygonToCellsNullPolygon_full) {
+        int64_t numHexagons;
+        t_assertSuccess(H3_EXPORT(maxPolygonToCellsSizeExperimental)(
+            &nullGeoPolygon, 9, CONTAINMENT_FULL, &numHexagons));
+        t_assert(numHexagons == 0, "got expected estimated size");
+        H3Index *hexagons = calloc(numHexagons, sizeof(H3Index));
+
+        t_assertSuccess(H3_EXPORT(polygonToCellsExperimental)(
+            &nullGeoPolygon, 9, CONTAINMENT_FULL, hexagons));
+        int64_t actualNumIndexes = countNonNullIndexes(hexagons, numHexagons);
+
+        t_assert(actualNumIndexes == 0, "got expected polygonToCells size");
+        free(hexagons);
+    }
+
+    TEST(polygonToCellsNullPolygon_overlapping) {
+        int64_t numHexagons;
+        t_assertSuccess(H3_EXPORT(maxPolygonToCellsSizeExperimental)(
+            &nullGeoPolygon, 9, CONTAINMENT_OVERLAPPING, &numHexagons));
+        t_assert(numHexagons == 0, "got expected estimated size");
+        H3Index *hexagons = calloc(numHexagons, sizeof(H3Index));
+
+        t_assertSuccess(H3_EXPORT(polygonToCellsExperimental)(
+            &nullGeoPolygon, 9, CONTAINMENT_OVERLAPPING, hexagons));
+        int64_t actualNumIndexes = countNonNullIndexes(hexagons, numHexagons);
+
+        t_assert(actualNumIndexes == 0, "got expected polygonToCells size");
+        free(hexagons);
+    }
```

Comment:
maybe these could be collapsed to the same test and just loop through which flags to test? Might be easier to read that way?

Reply:
True, that's what you did in #800. I can collapse this one, and the null hole, but not the other two (where FULL behaves differently)

Reply:
Oh, actually this is the only one I can collapse, the others have different results per mode